lldb/source/Commands/CommandObjectBreakpoint.cpp

```diff
@@ -200,41 +200,1341 @@ class lldb_private::BreakpointOptionGroup : public OptionGroup {
+    bool m_hardware = false; // FIXME - this can go in the "modify" options.
+    FileSpecList m_files;
+    FileSpecList m_modules;
+  };
+
+protected:
+  void DoExecute(Args &command, CommandReturnObject &result) override {
+    Target &target =
+        m_dummy_options.m_use_dummy ? GetDummyTarget() : GetTarget();
+
+    BreakpointSP bp_sp;
+    Status error;
+    bp_sp = target.CreateScriptedBreakpoint(
+        m_python_class_options.GetName().c_str(), &(m_options.m_modules),
+        &(m_options.m_files), false, m_options.m_hardware,
+        m_python_class_options.GetStructuredData(), &error);
+    if (error.Fail()) {
+      result.AppendErrorWithFormat(
+          "Error setting extra exception arguments: %s", error.AsCString());
+      target.RemoveBreakpointByID(bp_sp->GetID());
+      return;
+    }
+
+    if (bp_sp) {
+      CopyOverBreakpointOptions(bp_sp, m_bp_opts,
+                                m_name_opts.GetBreakpointNames(), result);
+      Stream &output_stream = result.GetOutputStream();
+      bp_sp->GetDescription(&output_stream, lldb::eDescriptionLevelInitial,
+                            /*show_locations=*/false);
+      if (&target == &GetDummyTarget())
+        output_stream.Printf("Breakpoint set in dummy target, will get copied "
+                             "into future targets.\n");
+      else {
+        // Don't print out this warning for exception breakpoints.  They can
+        // get set before the target is set, but we won't know how to actually
+        // set the breakpoint till we run.
+        if (bp_sp->GetNumLocations() == 0) {
+          output_stream.Printf("WARNING:  Unable to resolve breakpoint to any "
+                               "actual locations.\n");
+        }
+      }
+      result.SetStatus(eReturnStatusSuccessFinishResult);
+    } else {
+      result.AppendError("Breakpoint creation failed: No breakpoint created.");
+    }
+  }
+
+private:
+  BreakpointOptionGroup m_bp_opts;
+  BreakpointNamesOptionGroup m_name_opts;
+  BreakpointDummyOptionGroup m_dummy_options;
+  OptionGroupPythonClassWithDict m_python_class_options;
+  CommandOptions m_options;
+  OptionGroupOptions m_all_options;
+};
+
+#pragma mark Add::CommandOptions
+#define LLDB_OPTIONS_breakpoint_add
+#include "CommandOptions.inc"
+
+#pragma mark Add
+
+class CommandObjectBreakpointAdd : public CommandObjectMultiword {
+public:
+  CommandObjectBreakpointAdd(CommandInterpreter &interpreter)
+      : CommandObjectMultiword(interpreter, "add",
+                               "Commands to add breakpoints of various types") {
+    SetHelpLong(
+        R"(
+Access the breakpoint search kernels built into lldb.  Along with specifying the
+search kernel, each breakpoint add operation can specify a common set of 
+"reaction" options for each breakpoint.  The reaction options can also be
+modified after breakpoint creation using the "breakpoint modify" command.       
```

Comment:
extra spaces ? Is it `clang-format` who put these ?

```suggestion
Access the breakpoint search kernels built into lldb. Along with specifying the
search kernel, each breakpoint add operation can specify a common set of 
"reaction" options for each breakpoint. The reaction options can also be
modified after breakpoint creation using the "breakpoint modify" command.       
```

Reply:
In real English (as opposed to what you type into chat fields) there are always two spaces after a period.